Source/NSFileWrapper.m

```diff
@@ -44,325 +44,7 @@
-                              error: NULL];
-          while ((key = (NSString*)[enumerator nextObject]))
-            {
-              NSString *newPath =
-                  [path stringByAppendingPathComponent: key];
-              [[_wrapperData objectForKey: key] writeToFile: newPath
-                                                 atomically: atomicFlag
-                                         updateFilenames: updateFilenamesFlag];
-            }
-          success = YES;
-	  break;
-        }
-      case GSFileWrapperRegularFileType: 
-        {
-	  if ([_wrapperData writeToFile: path atomically: atomicFlag])
-	    success = [fm changeFileAttributes: _fileAttributes
-			  atPath: path];
-	  break;
-        }
-      case GSFileWrapperSymbolicLinkType: 
-        {
-          success = [fm createSymbolicLinkAtPath: path pathContent: _wrapperData];
-	  break;
-        }
-    }
-  if (success && updateFilenamesFlag)
-    {
-      [self setFilename: [path lastPathComponent]];
-    }
-  
-  return success;
-}
-
-- (NSData*) serializedRepresentation
-{
-  // FIXME - This should use a serializer. To get that working a helper object 
-  // is needed that implements the NSObjCTypeSerializationCallBack protocol.
-  // We should add this later, currently the NSArchiver is used.
-  // Thanks to Richard, for pointing this out.
-  return [NSArchiver archivedDataWithRootObject: self]; 
-}
-
-- (void) setFilename: (NSString*)filename
-{
-  if (filename == nil || [filename isEqualToString: @""])
-    {
-      [NSException raise: NSInternalInconsistencyException
-		   format: @"Empty or nil argument to setFilename: "];
-    }
-  else
-    {
-      ASSIGN(_filename, filename);
-    }
-}
-
-- (NSString*) filename
-{
-  return _filename;
-}
-
-- (void) setPreferredFilename: (NSString*)filename
-{
-  if (filename == nil || [filename isEqualToString: @""])
-    {
-      [NSException raise: NSInternalInconsistencyException
-		   format: @"Empty or nil argument to setPreferredFilename: "];
-    }
-  else
-    {
-      ASSIGN(_preferredFilename, filename);
-    }
-}
-
-- (NSString*) preferredFilename
-{
-  return _preferredFilename;
-}
-
-- (void) setFileAttributes: (NSDictionary*)attributes
-{
-  if (_fileAttributes == nil)
-    {
-      _fileAttributes = [NSMutableDictionary new];
-    }
-  
-  [_fileAttributes addEntriesFromDictionary: attributes];
-}
-
-- (NSDictionary*) fileAttributes
-{
-  return _fileAttributes;
-}
-
-- (BOOL) isRegularFile
-{
-  if (_wrapperType == GSFileWrapperRegularFileType)
-    {
-      return YES;
-    }
-  else
-    { 
-      return NO;
-    }
-}
-
-- (BOOL) isDirectory
-{
-  if (_wrapperType == GSFileWrapperDirectoryType)
-    {
-      return YES;
-    }
-  else
-    {
-      return NO;
-    }
-}
-
-- (BOOL) isSymbolicLink
-{
-  if (_wrapperType == GSFileWrapperSymbolicLinkType)
-    {
-      return YES;
-    }
-  else
-    {
-      return NO;
-    }
-}
+@implementation NSFileWrapper (NSExtensions)
```

Comment:
The name of the instance variable has been changed back to _icon, this needs to be changed here. This only works as this file has still the old name and never gets compiled.

Reply:
Just to be clear.... it was never ever changed TO _iconImage. :) Certainly not by me.  I changed it to _icon since you suggested it.

Reply:
This has been changed.  I think you caught me in the middle of a change here.